scrape/scrape.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import annotations
+from exceptions import NeedYearArgumentOnlyException, NonNumericYearArgumentException
+from typing import Dict, Union, List, Tuple
+from bs4 import BeautifulSoup
+from calendar_dict import INT_TO_MONTH, INT_TO_WEEKDAY_DICT
+
+import requests
+import json
+import sys
+import os
+import datetime
+import time
+import bs4
+
+# Example: python scrape.py 2080 fetches data for 2080B.S and saves them in database/2080.json
+
+def fetch_and_parse(year : int) -> Dict[str, Dict]:
+
+    HAMRO_PATRO_URL : str = f"https://www.hamropatro.com/calendar/{year}/"
+
+    current_month : int = 1
+    max_month : int = 12
+
+    all_months_data : Dict[str, Dict] = {}
+
+    while current_month <= max_month:
+
+        current_month_dict : Dict[str, Union[str,Dict]] = {}
+        all_months_data[INT_TO_MONTH[current_month]] = current_month_dict
+
+        r : requests.models.Response = requests.get(url=HAMRO_PATRO_URL+str(current_month))
```

Comment:
Handle exception properly when the year is invalid like negative number or `1000` or something like that… In general, you should check if the response status code is not 200.

Reply:
The negative year values are handled by the `isnumeric` string method. https://github.com/sumanchapai/patro/blob/72bc85b4032c835f26e2d5d48b277987437f8157/scrape/scrape.py#L111-L112
However, I will modify to check for other invalid year values with response status code which I agree is something that needs to be handled.